gpflow/models/gpmc.py

```diff
@@ -24,10 +24,10 @@
 from ..likelihoods import Likelihood
 from ..mean_functions import MeanFunction
 from ..utilities import to_default_float
-from .model import RegressionData, GPModel, MeanAndVariance, BayesianModelStoringData
+from .model import RegressionData, MeanAndVariance, InternalDataGPModel
 
 
-class GPMC(GPModel, BayesianModelStoringData):
+class GPMC(InternalDataGPModel):
```

Comment:
Note that for GPMC (and SGPMC in sgpmc.py) I explicitly re-defined training_loss - not because the code is different but because it's conceptually different and I wanted to highlight that... don't know how that interacts with your changes here?

Reply:
If you reimplement in the derived it wont interact badly, though i dont know whether we can call the density a `loss`. Its very possible having `training loss` with `GPMC` and `SGPMC` makes no sense, and they should just inherit from `GPModel`

Reply:
we don't call the density a loss, but you might still want to do some gradient optimization first to relax the model before starting MCMC - that's why I chose to overwrite it